### PR TITLE
UI: add exec handling for dead jobs/task states

### PR DESCRIPTION
**Pull request by backspace**
Here’s a screenshot what happens when a task has become inactive and you try to submit the exec command:

image

Here’s what it looks like when the job is dead:

image

I couldn’t get the automated test for an inactive task to work so there’s a `skip`ed placeholder for now 😞

### CODE REVIEW

**Comment by backspace** [comment]
This has #7499 as a base branch but it could be `master` if/when it’s merged.

**Comment by backspace** [comment]
(fixing merge conflicts 😳)

File: ui/app/templates/exec.hbs [hunk 1]
```diff
@@ -29,20 +29,30 @@
...
-      {{/each}}
-    </ul>
+{{#if (eq model.status "dead")}}
+  <div class="tree-and-terminal" data-test-exec-job-dead>
+    <div class="task-group-tree">
+    </div>
+    <div class="terminal-container" data-test-exec-job-dead-message>
+      Job <code>{{model.name}}</code> is dead and cannot host an exec session.
+    </div>
+  </div>
```
Comment (DingoEatingFuzz) [inline ui/app/templates/exec.hbs:39]: We can probably use space better with this empty/dead state, but any communication is better than a generic socket error message.

Comment (backspace) [reply ui/app/templates/exec.hbs:39]: ya agreed, I wish I’d thought of this during the design phase 😞

**Pull request merged**